packages/spectacular/src/lib/application-testing/app-component/spectacular-app.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, ViewChild } from '@angular/core';
+import { RouterOutlet } from '@angular/router';
+
+export const spectacularAppTagName = 'spectacular-app';
+
+@Component({
+  selector: spectacularAppTagName,
+  template: '<router-outlet></router-outlet>',
+})
+export class SpectacularAppComponent {
+  @ViewChild(RouterOutlet)
+  routerOutlet?: RouterOutlet;
+
+  getActiveComponent<TActiveComponent>(): TActiveComponent {
```

Comment:
A convention to make it easier to identify type parameters. Let's have a vote in the team.

Reply:
@NachoVazquez @santoshyadavdev we are waiting for you votes :)

Reply:
Vote for: [Yes]

Reply:
Vote for: [No]

Reply:
Sorry, I made a comment about this in the Discord channel. 

For me it is easier to identify when a type is a generic type using the convention, however, I understand that you could figure it out by context.

I don't have a strong opinion, but I maybe keep it, BTW Alex doesn't mention anything about that in its new style guide.

Reply:
Is there any resources to write the generics like this: `TFoo`?

Reply:
https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/generic-type-naming.md and every resource on Java and .NET.

Reply:
@SerkanSipahi It's not clear what Yes and No mean in your vote.

Reply:
Prefixing with T will also prevent name clashing with actual types.

Reply:
Given this information, I would vote to keep the T prefix.

Reply:
Updated the votes.

Reply:
React with thumbs up if you want to keep the T prefix, react with thumbs down if you want to take it out.

Reply:
I am not 100% sure but this does not conform to the Typescript convention. What is recommended is T, R, etc or full description of generic names. There is nothing against full names because I read that somewhere by Daniel Rosenwasser in a tweet.

Should we really let Java and Net influence us?

Reply:
Here is a long discussion at Microsoft.

https://docs.microsoft.com/en-us/archive/blogs/kcwalina/design-guidelines-update-names-of-generic-type-parameters

Reply:
This is from 2004 and I'm not sure if it applies to Typescript as well. The same is applied for interfaces where it is recommended to put an "I" in front of each interface name but here Typescript follows its own style guide see link. The other link is unfortunately in german but there it is discussed to put an "I" in front in Net.

typescript: https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#interface
net: https://docs.microsoft.com/de-de/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces

Which standard should we follow?

Reply:
The TypeScript handbook just suggests `T` which we can probably agree is a bad practice. I looked into the TypeScript, Angular, NgRx, and RxJS codebases. Unfortunately, it seems like they are all stuck on single letter type parameter names, usually just `T`.

I have presented arguments for keeping it. I haven't seen an argument for removing it yet. It's either a 2/2 or a 3/1 vote in favor of keeping the `T` prefix, so we'll leave it for now. If you want to change an organization convention for us, raise an issue in our chat or in one of our repos and make your arguments. I don't think further discussion adds value to this PR. If it's okay with you, let's resolve this discussion so that we can make progress with this feature.

Reply:
Sure 🙂

Reply:
Thank you, @SerkanSipahi.